### PR TITLE
Issue 2219 initial soc from experiment

### LINKED ISSUE #2219: Calculate `initial_soc` from model states
### Description

If I understand correctly the `initial_soc` is initialised from the parameter values. It would be good to initialise it from the model states (if the model initialised from solution).

### Motivation

This would allow for experiments such as RPT, in which we want to start from a given state (e.g. ageing) but run the simulation from 100% SoC.

### Possible Implementation

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DrSOKane**
# Description

Added new rpt-experiment notebook to demonstrate how to set up degradation experiments with reference performance tests (RPTs).

Fixes #2219 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all`
- [x] The documentation builds: `$ python run-tests.py --doctest`

You can run unit and doctests together at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by rtimms** [comment]
@DrSOKane @ganmad do you know what happened to the version of this notebook with the additional custom plots (can’t remember exactly what they were)? that would be a nice addition

**Comment by DrSOKane** [comment]
> @DrSOKane @ganmad do you know what happened to the version of this notebook with the additional custom plots (can’t remember exactly what they were)? that would be a nice addition

Do you mean this one by Ganesh? https://github.com/ganmad/PyBaMM-Local/blob/issue-2219-initial-SoC-from-experiment/examples/notebooks/rpt-experiment.ipynb

**Comment by rtimms** [comment]
yes, thanks! would be good to add those with some explanation

**Comment by ganmad** [comment]
Hi Rob and Simon,

Thanks for reminding. Will work on it and add some explanation on it. Happy
to receive support on this.

Best regards,
Ganesh

File: docs/source/examples/notebooks/rpt-experiment.ipynb [hunk 1]
```diff
@@ -0,0 +1,424 @@
+{
```
Comment (brosaplanella) [inline docs/source/examples/notebooks/rpt-experiment.ipynb:1]: <p>A few minor style comments:</p><ul><li>Use bullet points (<code>*</code> in markdown)</li><li>Format the variables (e.g. <code>cccv_experiment</code>) as inline code, using <code>`</code> before and after the variable.</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pybamm-team/PyBaMM/pull/2851/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

**Review by brosaplanella — APPROVED** [review]
Thanks Simon! Looks good to me, just a small style comment. It would be good to get @rtimms review before merging.

**Comment by rtimms** [comment]
Thanks, @DrSOKane looks good! I think the notebook text could be updated to give a better explanation of what is being done and why, but otherwise good to go!

**Review by rtimms — APPROVED** [review]

**Pull request merged**